src/renderers/WebGLRenderer.js

```diff
@@ -1500,10 +1500,14 @@ function WebGLRenderer( parameters ) {
 
 			material.onBeforeCompile( materialProperties.shader, _this );
 
+			parameters.shaderName = materialProperties.shader.name;
+			parameters.vertexShader = materialProperties.shader.vertexShader;
+			parameters.fragmentShader = materialProperties.shader.fragmentShader;
+
```

Comment:
I think it would be nice if this code section is part of `WebGLProgram.getParameters()`. In this way, `parameters` is not changed within `initMaterial()`.

Reply:
I changed onbeforecompile to run before getting parameters